### PR TITLE
Add create id set step

### PR DESCRIPTION
**Pull request by ChanochShayner**
Add create id set step

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -51,15 +51,35 @@ jobs:
...
+        gsutil -m cp -r gs://marketplace-dist/content/private_id_set.json artifacts/private_id_set.json
+        gsutil -m cp -r gs://marketplace-dist/content/id_set.json artifacts/id_set.json
+
+        python3 tools/prepare_private_id_set_for_merge.py -pis artifacts/id_set.json
+        python3 tools/prepare_private_id_set_for_merge.py -pis artifacts/private_id_set.json
+
+        demisto-sdk merge-id-sets -i1 artifacts/pack_id_set.json -i2 artifacts/id_set.json -o artifacts/id_set.json
+        demisto-sdk merge-id-sets -i1 artifacts/private_id_set.json -i2 artifacts/id_set.json -o artifacts/id_set.json
+    - name: Archive ID set file
+      if: ${{ always() }}
```
Comment (guykeller) [inline .github/workflows/build.yml]:
```suggestion
      if: always()
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -51,15 +51,35 @@ jobs:
...
+
+        gsutil -m cp -r gs://marketplace-dist/content/private_id_set.json artifacts/private_id_set.json
+        gsutil -m cp -r gs://marketplace-dist/content/id_set.json artifacts/id_set.json
+
+        python3 tools/prepare_private_id_set_for_merge.py -pis artifacts/id_set.json
+        python3 tools/prepare_private_id_set_for_merge.py -pis artifacts/private_id_set.json
+
+        demisto-sdk merge-id-sets -i1 artifacts/pack_id_set.json -i2 artifacts/id_set.json -o artifacts/id_set.json
+        demisto-sdk merge-id-sets -i1 artifacts/private_id_set.json -i2 artifacts/id_set.json -o artifacts/id_set.json
+    - name: Archive ID set file
```
Comment (guykeller) [inline .github/workflows/build.yml]: archive? you seem to be uploading it as artifacts

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -51,15 +51,35 @@ jobs:
...
+      if: ${{ always() }}
+      uses: actions/upload-artifact@v1
+      with:
+        name: id_set
+        path: artifacts/id_set.json
     - name: Validate all packs
       run: |
-        demisto-sdk validate -a
+        git fetch
+        git branch
```
Comment (guykeller) [inline .github/workflows/build.yml:73]: why?

File: .github/workflows/build.yml (hunk 1 above)
Comment (guykeller) [reply .github/workflows/build.yml]: change this everywhere

File: tools/prepare_private_id_set_for_merge.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import json
```
Comment (guykeller) [inline tools/prepare_private_id_set_for_merge.py]: is this a copy of an existing script?

**Review by guykeller — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (ChanochShayner) [reply .github/workflows/build.yml]: done

File: .github/workflows/build.yml (hunk 2 above)
Comment (ChanochShayner) [reply .github/workflows/build.yml]: done

File: .github/workflows/build.yml (hunk 3 above)
Comment (ChanochShayner) [reply .github/workflows/build.yml:73]: For a new test added in sdk
@yaakovi @roysagi

File: tools/prepare_private_id_set_for_merge.py (hunk 4 above)
Comment (ChanochShayner) [reply tools/prepare_private_id_set_for_merge.py]: There is another script that is similar to it - but they are different in many ways

File: .github/workflows/build.yml (hunk 3 above)
Comment (ChanochShayner) [reply .github/workflows/build.yml:73]: @yaakovi Should it be added to private build as well?

**Comment by ChanochShayner** [comment]
@anara123 Do we want to upload the ID set to artifacts?
It's not needed for the commands in build, but it might be worth it in case we need to debug it.

File: tools/prepare_private_id_set_for_merge.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    parser.add_argument('-pis', '--private_id_set_path', help='Private ID set path', required=True)
+
+    options = parser.parse_args()
+    return options
+
+
+def main():
+    options = options_handler()
+    private_id_set_path = options.private_id_set_path
+    new_pack_name = options.new_pack_name
```
Comment (anara123) [inline tools/prepare_private_id_set_for_merge.py]: this functionality should be as part of the merge-id-sets command.

**Review by anara123 — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -51,15 +51,35 @@ jobs:
...
         path: artifacts/lint_debug_log.log
+    - name: Create ID Set
+      run: |
+        demisto-sdk create-id-set -o artifacts/pack_id_set.json
+
+        gsutil -m cp -r gs://marketplace-dist/content/private_id_set.json artifacts/private_id_set.json
+        gsutil -m cp -r gs://marketplace-dist/content/id_set.json artifacts/id_set.json
+
+        python3 tools/prepare_private_id_set_for_merge.py -pis artifacts/id_set.json
+        python3 tools/prepare_private_id_set_for_merge.py -pis artifacts/private_id_set.json
```
Comment (anara123) [inline .github/workflows/build.yml]: remove these liness

File: tools/prepare_private_id_set_for_merge.py (hunk 5 above)
Comment (ChanochShayner) [reply tools/prepare_private_id_set_for_merge.py]: done

File: .github/workflows/build.yml (hunk 6 above)
Comment (ChanochShayner) [reply .github/workflows/build.yml]: done

File: .github/requirements.txt [hunk 7]
```diff
@@ -1 +1 @@
-demisto-sdk
+git+https://github.com/demisto/demisto-sdk.git@merge-id-sets_add_duplication_check
```
Comment (anara123) [inline .github/requirements.txt]: don't forget to change it back

**Review by anara123 — APPROVED** [review]
looks good. And see @guykeller comment

File: .github/requirements.txt (hunk 7 above)
Comment (ChanochShayner) [reply .github/requirements.txt]: done

**Review by guykeller — APPROVED** [review]

**Pull request merged**